helpers/report_exports.py

```
@@ -0,0 +1,858 @@
+    try:
+        metadata = os.stat(
+            filename,
+            dir_fd=directory_fd,
+            follow_symlinks=False,
+        )
+    except FileNotFoundError:
+        if allow_absent:
+            return None
+        raise ReportExportError(
+            f"Owned {artifact_label} changed before operation: {filename}"
+        )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)